src/lib.js

```diff
@@ -265,6 +265,103 @@ function ByteQueue(size)
 }
 
 
+/**
+ * @constructor
+ *
+ * Queue wrapper around Float32Array
+ * Used by devices such as the sound blaster sound card
+ */
+function FloatQueue(size)
+{
+    var data = new Float32Array(size),
+        start,
+        end;
```

Comment:
For new code, use prototype and properties instead of closures and variables (like the `CircularQueue` below).